File: TemplateTools/src/main/java/com/lh/component/common/Point.java
package com.lh.component.common;

public class Point implements pl.luwi.series.reducer.Point {
    private Pair<Double, Double> mPoint;
    private static final double FLOAT_ERROR = 0.00001f;

    public Point(double x, double y) {
        mPoint = new Pair<>(x, y);
    }

    public void setPosition(double newX, double newY) {
        mPoint = new Pair<>(newX, newY);
    }

    public double x() {
        return mPoint.first;
    }

    public double y() {
        return mPoint.second;
    }

    @Override
    public boolean equals(Object o) {
        if (o instanceof Point) {
            return this.x() - ((Point) o).x() <= FLOAT_ERROR && this.y() - ((Point) o).y() <= FLOAT_ERROR;
        }
        return super.equals(o);
    }

    @Override
    public double getX() {
        return x();
    }

    @Override
    public double getY() {
        return y();
    }
}

File: Keyboard/app/src/main/java/com/lh/kete/algorithm/common/Point.kt
package com.lh.kete.algorithm.common

import pl.luwi.series.reducer.Point

class Point : Point {
    val mX: Float
    val mY: Float

    override fun getX(): Double {
        return mX.toDouble()
    }

    override fun getY(): Double {
        return mY.toDouble()
    }

    constructor(x: Float, y: Float) {
        this.mX = x
        this.mY = y
    }

    override fun toString(): String {
        return String.format("(%f,%f)", mX, mY)
    }
}